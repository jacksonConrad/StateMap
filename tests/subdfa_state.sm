// Simple test to check that a subdfa can be called as a function


void DFA sub_dfa1() {
	start {
		print(state("sub_dfa2"));
		red <- *;
	}

	red {
		print(state("sub_dfa2"));
		blue <- *;
	}

	blue {
		print(state("sub_dfa2"));
		return;
	}
}

void DFA sub_dfa2() {
	start {
		print(state("sub_dfa1"));
		red <- *;
	}

	red {
		print(state("sub_dfa1"));
		blue <- *;

	}

	blue {
		print(state("sub_dfa1"));
		return;
	}
}

void DFA main()	
{
	start	
	{
	concurrent(sub_dfa1(), sub_dfa2());
	return;
	}
}

