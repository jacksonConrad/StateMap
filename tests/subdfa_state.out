start
start
red
red
blue
blue
